### PR TITLE
Wasmd v0.27.0-rc0 integration

### LINKED ISSUE #239: Decide on max_wasm_code_size
Traditionally we used 600 KiB which was changed to 1200 KiB in https://github.com/CosmWasm/wasmd/pull/550. I am not aware of any testing or analysis what the implications of this new value is. It now made it into Tgrade.

I think we should double check if 1200 KiB is really the value we want to offer in the first version of Tgrade. Those magabytes we will have to carry around for a long time and it will be very hard to decrease the limit later on.

Terra is talking about increasing the limit from 600 -> 1200 but the arguments don't seem to be very convincing on both sides:
- https://station.terra.money/proposal/179
- https://agora.terra.money/t/increase-the-wasm-code-size-in-terra-blockchain/3472
- https://twitter.com/SmartStake/status/1484556009842659329

### PR DESCRIPTION
**Pull request by alpe**
* Resolves #239

Adds
* Statesync
* Interchain accounts (host role)
* Wasmd v0.27

### CODE REVIEW

File: app/ante.go [hunk 1]
```diff
@@ -29,7 +28,7 @@ type HandlerOptions struct {
 	SignModeHandler authsigning.SignModeHandler
 	SigGasConsumer  func(meter sdk.GasMeter, sig signing.SignatureV2, params authtypes.Params) error
 
-	IBCChannelkeeper  channelkeeper.Keeper
+	IBCCoreKeeper     *ibcCoreKeeper.Keeper
```
Comment (ethanfrey) [inline app/ante.go:31]: looks good

File: app/app.go [hunk 2]
```diff
@@ -179,12 +186,14 @@ type TgradeApp struct {
 	ibcKeeper        *ibckeeper.Keeper // IBC Keeper must be a pointer in the app, so we can SetRouter on it correctly
 	evidenceKeeper   evidencekeeper.Keeper
 	transferKeeper   ibctransferkeeper.Keeper
+	icaHostKeeper    icahostkeeper.Keeper
```
Comment (ethanfrey) [inline app/app.go:189]: Once wasmd #848 is merged, you can add controller functionality as well

File: app/app.go [hunk 3]
```diff
@@ -550,6 +582,23 @@ func NewTgradeApp(
 	app.SetBeginBlocker(app.BeginBlocker)
 	app.SetEndBlocker(app.EndBlocker)
 
+	// must be before Loading version
+	// requires the snapshot store to be created and registered as a BaseAppOption
+	// see cmd/wasmd/root.go: 206 - 214 approx
+	if manager := app.SnapshotManager(); manager != nil {
```
Comment (ethanfrey) [inline app/app.go:588]: Very nice

**Review by ethanfrey — COMMENTED** [review]
Looks good to match with v0.27.0-rc0.

It would be nice to

1. merge open wasmd PR
2. tag wasmd v0.27.0-rc2
3. update this to use that (and ICA controller)

before merging this. I assume -rc2 will basically be the final version.

Also, I didn't see the code where you modify the max wasm size constant. I was assuming you would set that global constant from x/wasm in app.go, but maybe you did it somewhere else and I missed it.

**Comment by alpe** [comment]
>  I was assuming you would set that global constant ..

We go with the new wasmd default #239 . I have updated the description for this issue

**Review by ethanfrey — APPROVED** [review]
Let's get this in, and update to the v0.27.0 when that is out.

**Pull request merged**